gabarit/template_nlp/nlp_project/package_name/utils.py

```diff
@@ -319,12 +349,10 @@ def find_folder_path(folder_name: str, base_folder: Union[str, None] = None) ->
 class NpEncoder(json.JSONEncoder):
     '''JSON encoder to manage numpy objects'''
     def default(self, obj) -> Any:
-        if isinstance(obj, np.integer):
-            return int(obj)
-        elif isinstance(obj, np.floating):
-            return float(obj)
-        elif isinstance(obj, np.ndarray):
-            return obj.tolist()
+        if is_ndarray_convertable(obj):
+            return ndarray_to_builtin_object(obj)
+        elif isinstance(obj, set):
+            return list(obj)
```

Comment:
Was it needed ?

Reply:
To be more generic. sets are note JSON serializable.